pkg/reconciler/common/releases_test.go

```diff
@@ -980,3 +988,96 @@ func TestTargetManifestPathArray(t *testing.T) {
 		})
 	}
 }
+
+func TestInstalledManifest(t *testing.T) {
+	koPath := "testdata/kodata"
+	os.Setenv(KoEnvKey, koPath)
+	defer os.Unsetenv(KoEnvKey)
+
+	tests := []struct {
+		name                  string
+		component             v1alpha1.KComponent
+		expectedManifestsPath string
+	}{{
+		name: "knative-serving with the version and manifests in the status",
+		component: &v1alpha1.KnativeServing{
+			Spec: v1alpha1.KnativeServingSpec{
+				CommonSpec: v1alpha1.CommonSpec{
+					Version: "0.16.1",
+				},
+			},
+			Status: v1alpha1.KnativeServingStatus{
+				Version:   "0.16.1",
+				Manifests: []string{"testdata/kodata/knative-serving/0.16.1"},
+			},
+		},
+		expectedManifestsPath: "testdata/kodata/knative-serving/0.16.1",
+	}, {
+		name: "knative-serving with the version in the status",
+		component: &v1alpha1.KnativeServing{
+			Spec: v1alpha1.KnativeServingSpec{
+				CommonSpec: v1alpha1.CommonSpec{
+					Version: "0.16.1",
+				},
+			},
+			Status: v1alpha1.KnativeServingStatus{
+				Version: "0.16.0",
+			},
+		},
+		expectedManifestsPath: "testdata/kodata/knative-serving/0.16.0",
+	}, {
+		name: "knative-serving with multiple paths in status.manifests",
+		component: &v1alpha1.KnativeServing{
+			Spec: v1alpha1.KnativeServingSpec{
+				CommonSpec: v1alpha1.CommonSpec{
+					Version: "0.16.1",
+				},
+			},
+			Status: v1alpha1.KnativeServingStatus{
+				Version: "0.16.1",
+				Manifests: []string{"testdata/kodata/knative-serving/0.16.1/serving-crd.yaml" + "," +
+					"testdata/kodata/knative-serving/0.16.1/serving-core.yaml",
+					"testdata/kodata/additional-manifests/additional-resource.yaml"},
+			},
+		},
+		expectedManifestsPath: "testdata/kodata/knative-serving/0.16.1/serving-crd.yaml" + "," +
+			"testdata/kodata/knative-serving/0.16.1/serving-core.yaml" + "," +
+			"testdata/kodata/additional-manifests/additional-resource.yaml",
+	}, {
+		name: "knative-serving with status.version unavailable in kodata",
```

Comment:
Out of curious, this case (Even if we set spec.version to 0.16.1, status becomes 0.12.0) does happen?

Reply:
It is possible.
We have 0.12.0 knative serving installed. After a long time, users installed the latest version of the operator. They changed the spec.version into 0.16.1, and apply the CR. The operator will failed the validation by providing the error message in the CR, that unable to migrate across multiple minor versions.